src/screens/Airdrops/utils.ts

```diff
@@ -0,0 +1,282 @@
+import { TransactionReceipt, TransactionRequest } from '@ethersproject/abstract-provider';
+import { Wallet } from '@ethersproject/wallet';
+import { Address, formatUnits } from 'viem';
+import { analyticsV2 } from '@/analytics';
+import { NativeCurrencyKey, NewTransaction, TransactionStatus } from '@/entities';
+import { LedgerSigner } from '@/handlers/LedgerSigner';
+import { getProvider } from '@/handlers/web3';
+import showWalletErrorAlert from '@/helpers/support';
+import { add, convertAmountToNativeDisplayWorklet, multiply, formatNumber } from '@/helpers/utilities';
+import { logger, RainbowError } from '@/logger';
+import { loadWallet } from '@/model/wallet';
+import { weiToGwei } from '@/parsers';
+import { RainbowClaimable } from '@/resources/addys/claimables/types';
+import { isStaging } from '@/resources/addys/client';
+import { lessThanOrEqualToWorklet } from '@/safe-math/SafeMath';
+import { userAssetsStore } from '@/state/assets/userAssets';
+import { useBackendNetworksStore } from '@/state/backendNetworks/backendNetworks';
+import { ChainId } from '@/state/backendNetworks/types';
+import { getNextNonce } from '@/state/nonces';
+import { addNewTransaction } from '@/state/pendingTransactions';
+import { GasSettings } from '@/__swaps__/screens/Swap/hooks/useCustomGas';
+import { safeBigInt } from '@/__swaps__/screens/Swap/hooks/useEstimatedGasFee';
+import { calculateGasFeeWorklet } from '@/__swaps__/screens/Swap/providers/SyncSwapStateAndSharedValues';
+import { time } from '@/utils';
+
+export interface GasInfo {
+  gasFeeDisplay: string | undefined;
+  gasLimit: string | undefined;
+  sufficientFundsForGas: boolean;
+}
+
+interface ClaimableTransactionData {
+  airdropId: string;
+  amount: string;
+  chainId: ChainId;
+  data: string;
+  symbol: string;
+  to: Address;
+  usdValue: number;
+}
+
+/**
+ * Executes an airdrop claim transaction.
+ */
+export async function executeAirdropClaim({
+  accountAddress,
+  claimable,
+  gasLimit,
+  gasSettings,
+  onConfirm,
+}: {
+  accountAddress: string;
+  claimable: RainbowClaimable;
+  gasLimit: string;
+  gasSettings: GasSettings;
+  onConfirm?: (receipt: TransactionReceipt) => void;
+}): Promise<{ error?: string; success: boolean; txHash?: string }> {
+  const { airdropId, amount, chainId, data, symbol, to, usdValue } = getClaimableTransactionData(claimable);
+  const nonce = await getNextNonce({ address: accountAddress, chainId });
+
+  const txPayload: TransactionRequest = {
+    chainId,
+    data,
+    from: accountAddress,
+    gasLimit: toHex(gasLimit),
+    nonce,
+    to,
+    value: '0x0',
+  };
```

Comment:
we have https://github.com/rainbow-me/rainbow/blob/04bf1e9e623398319d812f4d8845418021f5fccd/src/helpers/buildTransaction.ts#L50 if you want to utilize it here.